Add service for regridding zarrs

brews opened the pull request:
Adds service to regrid zarr files with `xesmf`.

Basic interface from CLI is 
```bash
dodola regrid /path/to/store.zarr \
    --out /path/to/regrid.zarr \
    --method "bilinear"
```
Can use `-r`/`--targetresolution` to set the target global grid resolution (1.0 x 1.0 is default). Can use `-w`/`--weightspath` to use to any precalculated regridding weights in a NetCDF file on disk.

Includes integration test to ensure regridding works with pre-computed regridding weights from `dodola buildweights`.

Close #27.

dgergel reviewed (changes requested):
@brews this looks great, just a few very small nitpicky comments. Thanks for putting this together so quickly!

dgergel commented on dodola/cli.py:
```diff
@@ -112,3 +112,39 @@ def rechunk(x, variable, chunk, maxmemory, out):
+    help="Local path to existing regrid weights file",
+)
+def regrid(x, out, method, targetgrid, weightspath):
+    """Regrid a target climate dataset
```
Actually nm - I see that we're using this in other places in `dodola` as well. I think it's fine for now.

brews commented on dodola/tests/test_services.py:
```diff
@@ -150,3 +150,118 @@ def test_rechunk():
+def test_regrid_methods(regrid_method, expected_shape):
+    """Smoke test that services.regrid outputs with different regrid methods
+
+    The the expected shape is the same, but change in methods should not error.
```
Got it. Thanks!

brews commented on dodola/tests/test_services.py:
```diff
@@ -150,3 +150,118 @@ def test_rechunk():
+    """Smoke test that services.regrid outputs with different regrid methods
+
+    The the expected shape is the same, but change in methods should not error.
+    """
```
Fixed. Thank you.

dgergel reviewed (approved):
LGTM!

brews commented:
Thanks for the review @dgergel. I appreciate the corrections. This PR is ready for re-review. I think I've addressed all of your concerns in the revision.

I know you wanted this merged ASAP, so feel free to merge this if you have no further corrections.

Thanks again!
